CDCgov/prime-reportstream
# Platform/angela/14257/circular json

This PR ...

Prevents malformed JSON that loops endlessly

Test Steps:
1. Create file ./junk/CircularJson/Test.hl7
2. Copy cleanHL7Record from UniversalPipelineTestUtils.kt into the file you created
3. Run `./gradlew primeCLI --args="fhirdata -i ./junk/CircularJson/Test.hl7 -o ./junk/CircularJson/SRApplied.fhir --output-format=FHIR -s classpath:/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml"`
4. Verify that no error occurs due to JSON nesting limit being exceeded

## Changes
- Use a copy of the value when setting a bundle property in the transform to prevent circular JSON
- Created a test sender transform to use that extends original-pipeline-transforms.yml
- Added two exceptions to gitleaks-config to prevent false positives being flagged in FHIR data in UniversalPipelineTestUtils.kt
- Updated test data in UniversalPipelineTestUtils.kt to include the extension being added by the test sender transform
- Updated FHIRConverterIntegrationTests from using original-pipeline-transforms to using the new test transform

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #14257
